### PR TITLE
Unit tests for couchbase/couchbase-lite-java-core#599

### PR DESCRIPTION
**Pull request by sergio91pt**
Related to issue couchbase/couchbase-lite-java-core#592

### CODE REVIEW

File: src/androidTest/java/com/couchbase/lite/AttachmentsTest.java [hunk 1]
```diff
@@ -829,4 +834,49 @@ public int read() throws IOException {
...
+        UnsavedRevision rev = doc.createRevision();
+        rev.setAttachment(attachmentName, "text/html", new ByteArrayInputStream(contentGzipped));
+        rev.save();
+
+        SavedRevision savedRev = doc.getCurrentRevision();
+        Attachment attachment = savedRev.getAttachment(attachmentName);
+
+        // As far as revision users are concerned their data is not gzipped
+        assertFalse(attachment.getGZipped());
+        MoreAsserts.assertEquals(content, IOUtils.toByteArray(attachment.getContent()));
```
Comment (hideki) [inline src/androidTest/java/com/couchbase/lite/AttachmentsTest.java]: Couchbase Lite unit tests are common code for Android and J2SE (Java). `MoreAsserts` is Android unit test. Please replace with common solution for both Android and J2SE.

File: src/androidTest/java/com/couchbase/lite/AttachmentsTest.java [hunk 2]
```diff
@@ -829,4 +834,49 @@ public int read() throws IOException {
...
+        assertFalse(attachment.getGZipped());
+
+        // But the it may be gzipped encoded internally
+        long sequence = savedRev.getSequence();
+        attachment = database.getAttachmentForSequence(sequence, attachmentName);
+        assertTrue(attachment.getGZipped());
+        MoreAsserts.assertEquals(contentGzipped, IOUtils.toByteArray(attachment.getContent()));
```
Comment (hideki) [inline src/androidTest/java/com/couchbase/lite/AttachmentsTest.java]: Couchbase Lite unit tests are common code for Android and J2SE (Java). `MoreAsserts` is Android unit test. Please replace with common solution for both Android and J2SE.

File: src/androidTest/java/com/couchbase/lite/AttachmentsTest.java [hunk 3]
```diff
@@ -829,4 +834,49 @@ public int read() throws IOException {
...
+        rev.setAttachment(attachmentName, "text/html", new ByteArrayInputStream(contentGzipped));
+        rev.save();
+
+        SavedRevision savedRev = doc.getCurrentRevision();
+        Attachment attachment = savedRev.getAttachment(attachmentName);
+
+        // As far as revision users are concerned their data is not gzipped
+        assertFalse(attachment.getGZipped());
+        MoreAsserts.assertEquals(content, IOUtils.toByteArray(attachment.getContent()));
+        assertFalse(attachment.getGZipped());
```
Comment (hideki) [inline src/androidTest/java/com/couchbase/lite/AttachmentsTest.java]: Same with Line 871

File: src/androidTest/java/com/couchbase/lite/AttachmentsTest.java [hunk 4]
```diff
@@ -829,4 +834,49 @@ public int read() throws IOException {
...
+
+        // But the it may be gzipped encoded internally
+        long sequence = savedRev.getSequence();
+        attachment = database.getAttachmentForSequence(sequence, attachmentName);
+        assertTrue(attachment.getGZipped());
+        MoreAsserts.assertEquals(contentGzipped, IOUtils.toByteArray(attachment.getContent()));
+        assertTrue(attachment.getGZipped());
```
Comment (hideki) [inline src/androidTest/java/com/couchbase/lite/AttachmentsTest.java]: Same with Line 879

**Comment by sergio91pt** [comment]
@hideki 
Done, can you review it again? I substituted MoreAsserts with assertTrue on Arrays.equals.

**Comment by hideki** [comment]
This one is already merged with https://github.com/couchbase/couchbase-lite-android/commit/67c974ef0693ab815d38605a41383fb0e70aea1c .
I don't know why this PR is not automatically closed.

**Comment by hideki** [comment]
@sergio91pt,
Thanks again for  your contributions!

**Comment by sergio91pt** [comment]
@hideki did you merge the commit to remove `MoreAsserts` 360f1fd3e074079f82d877f77666e5c57e0a66e3?

**Comment by hideki** [comment]
@sergio91pt,
Thank you for letting me know. I did local merge without pull your new commit.

**Pull request merged**